validator/lib/jsonapi/validator/resource.rb

```diff
@@ -0,0 +1,123 @@
+require 'jsonapi/validator/document'
+require 'jsonapi/validator/relationship'
+
+module JSONAPI
+  module Validator
+    class Resource
```

Comment:
I would personally prefer this to live in a separate gem, `strong_parameters`, something new, whatever. The rest of the gem is focused on generic, low-level jsonapi handling, this introduces a higher-level domain-specific usage that should live one level higher IMO.

I could also see this growing in complexity with type validation/coercion, rule matching, etc.